OpenSim/Examples/DataTable/example.cpp

```diff
@@ -0,0 +1,357 @@
+  OpenSim::DataTable_<SimTK::Vec6> dt_vec6{};
+
+  std::vector<SimTK::Real> data_real{};
+  std::vector<SimTK::Vec3> data_vec3{};
+  std::vector<SimTK::Vec6> data_vec6{};
+  for(int i = 0; i < 12; ++i) {
+    data_real.push_back(i);
+    data_vec3.push_back(SimTK::Vec3{1, 2, 3} + i);
+    data_vec6.push_back(SimTK::Vec6{1, 2, 3, 4, 5, 6} + i);
+  }
+
+  // Construct a DataTable using iterators. We can arrange the elements
+  // row-wise or column-wise using the argument 'dir'. The argument 'ndir' is
+  // required in either case to let the DataTable know about the shape of the
+  // matrix. The following creates a 4x3 DataTable filling the elements
+  // column-wise.
+  {
+    OpenSim::DataTable_<SimTK::Real> real{data_real.cbegin(),
+                                          data_real.cend(),
+                                          3,
+                                          OpenSim::COLWISE};
+    OpenSim::DataTable_<SimTK::Vec3> vec3{data_vec3.cbegin(),
+                                          data_vec3.cend(),
+                                          3,
+                                          OpenSim::COLWISE};
+    OpenSim::DataTable_<SimTK::Vec6> vec6{data_vec6.cbegin(),
+                                          data_vec6.cend(),
+                                          3,
+                                          OpenSim::COLWISE};
+  }
+
+  // Add multiple rows at once using an iterator. Because the DataTable is empty
+  // , number of columns has to specified when using iterator. When calling this
+  // function on a non-empty DataTable, the arugment ncol is ignored. We have 12
+  // elements in the iterator and following creates a 3x4 DataTable.
+  dt_real.addRows(data_real.cbegin(), data_real.cend(), 4);
+  dt_vec3.addRows(data_vec3.cbegin(), data_vec3.cend(), 4);
+  dt_vec6.addRows(data_vec6.cbegin(), data_vec6.cend(), 4);
+
+  std::vector<SimTK::Real> data1_real(4, 1);
+  std::vector<SimTK::Vec3> data1_vec3(4, {1, 2, 3});
+  std::vector<SimTK::Vec6> data1_vec6(4, {1, 2, 3, 4, 5, 6});
+
+  // Add a row to the DataTable using SimTK::RowVector_. Create SimTK::RowVector
+  // and add it as a row. After this the DataTable will of size 4x4.
+  SimTK::RowVector_<SimTK::Real> row_real{static_cast<int>(data1_real.size()), 
+                                          data1_real.data()};
+  SimTK::RowVector_<SimTK::Vec3> row_vec3{static_cast<int>(data1_vec3.size()), 
+                                          data1_vec3.data()};
+  SimTK::RowVector_<SimTK::Vec6> row_vec6{static_cast<int>(data1_vec6.size()), 
+                                          data1_vec6.data()};
+  dt_real.addRow(row_real);
+  dt_vec3.addRow(row_vec3);
+  dt_vec6.addRow(row_vec6);
+
+  // Add a single row to the DataTable using an iterator. Note that this 
+  // function is different from the one used above, which is addRows().
+  dt_real.addRow(data1_real.cbegin(), data1_real.cend());
```

Comment:
The idea is to not have any member functions that take a container as input. All such containers produce an input-iterator, which can be fed into an addRow overload.